# Домашнее задание

В домашнем задании вам необходимо рассмтреть данные с олимпиады НТИ по профилю "Исскусственный интеллект" за 2019-2020 год.

Задачу можно найти по ссылке:

https://www.kaggle.com/c/onti-students-performance/data

## **Как загрузить данные в Google Colab?**

1. Из Github (файлы <25 МБ)
2. С локального диска
3. С Google Диска

In [1]:
# Github
# 1. Нажмите на файл с данными, которые хотите подключить
# 2. Нажмите "Raw"
# 3. Скопируйте ссылку открывшейся старницы
import pandas as pd

url1 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/X_train.csv'
url2 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/studs_info.csv'
url3 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/zachety.csv'
url4 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/y_train.csv'

X_train = pd.read_csv(url1)
studs_info = pd.read_csv(url2)
zachety = pd.read_csv(url3)
y_train = pd.read_csv(url4)

In [ ]:
# С локального компьютера

from google.colab import files
uploaded = files.upload()

In [ ]:
studs_info = pd.read_csv('studs_info.csv')
X_train = pd.read_csv('X_train.csv')
y_train = pd.read_csv('y_train.csv')
zachety = pd.read_csv('zachety.csv')

In [ ]:
# Google disk

from google.colab import drive
drive.mount('/content/drive')

In [ ]:
path = "copied path"
# /content/drive/MyDrive/Data Science для школьников/ОНТИ /ONTI Students performance/X_train.csv
df_bonus = pd.read_csv(path)

## Вариант 1.

Кто делает:
1. Александров Данила Александрович
2. Быков Дмитрий Александрович 
3. Горбунов Сергей Михайлович 
4. Горячко Дарья Романовна 
5. Громов Алексей Игоревич
6. Игнатов Константин Дмитриевич
7. Ямпольская Ирина Вадимовна
8. Дидковский Илья Радионович

1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в получившемся датафрейме train.
4. Посчитать сколько раз встречается каждая специальность в колонке studs_info['направление (специальность)'].
5. Создать новый датафрейм из studs_info, в котором будут только женщины с лечебного дела.
6. Разложить колонку "Дата выпуска" на три: Год выпуска, Месяц выпуска, День выпуска.
7. Посчитать сколько уникальных значений в колонке studs_info['Категория обучения'].
8. Заменить в колонке studs_info['Категория обучения'] все строковые категории на числовые.
9. Удалить ненужную на ваш вгляд колонку из датафрейма studs_info.
10. Посчитать сколько мужчин выпустились 2019-08-31.

In [75]:
# 1

print(studs_info.shape)
print(X_train.shape)
print(y_train.shape)
print(zachety.shape)

(5466, 16)
(105058, 8)
(105058, 2)
(133223, 8)


In [95]:
X_train

Unnamed: 0  STD_ID             НАПРАВЛЕНИЕ        ГОД  КУРС  СЕМЕСТР  \
0            51081  175711     38.03.02 Менеджмент  2018-2019   2.0      4.0   
1            72921  100647  31.05.01 Лечебное дело  2015-2016   3.0      5.0   
2            96438  199311   31.05.03 Стоматология  2019-2020   1.0      1.0   
3           273044  162002     38.03.02 Менеджмент  2017-2018   2.0      3.0   
4           371309  108384  31.05.01 Лечебное дело  2018-2019   5.0      9.0   
...            ...     ...                     ...        ...   ...      ...   
105053      224856  176141  40.03.01 Юриспруденция  2017-2018   1.0      2.0   
105054      260998  182794  31.05.01 Лечебное дело  2018-2019   2.0      4.0   
105055       57245  162391  31.05.01 Лечебное дело  2016-2017   1.0      2.0   
105056      218529  202479      38.03.01 Экономика  2017-2018   1.0      1.0   
105057      233318  182325  08.03.01 Строительство  2018-2019   2.0      3.0   

                      АТТЕСТАЦИЯ  \
0                        Экзамен   
1                        Экзамен   
2                        Экзамен   
3                        Экзамен   
4                        Экзамен   
...                          ...   
105053           Курсовая работа   
105054                   Экзамен   
105055  Дифференцированный зачет   
105056  Дифференцированный зачет   
105057           Курсовая работа   

                                               ДИСЦИПЛИНА  
0                                               Маркетинг  
1       Пропедевтика внутренних болезней, лучевая диаг...  
2                                                   Химия  
3                                              Статистика  
4                                               Педиатрия  
...                                                   ...  
105053                              Конституционное право  
105054                              Нормальная физиология  
105055                                   Иностранный язык  
105056                                        Информатика  
105057           Технологические процессы в строительстве  

[105058 rows x 8 columns]

In [98]:
y_train

index  mark
0        51081     5
1        72921     5
2        96438     5
3       273044     4
4       371309     3
...        ...   ...
105053  224856     3
105054  260998     3
105055   57245     4
105056  218529     4
105057  233318     4

[105058 rows x 2 columns]

In [101]:
# 2 

train = pd.concat([X_train, y_train], 1)
train

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  This is separate from the ipykernel package so we can avoid doing imports until


Unnamed: 0  STD_ID             НАПРАВЛЕНИЕ        ГОД  КУРС  СЕМЕСТР  \
0            51081  175711     38.03.02 Менеджмент  2018-2019   2.0      4.0   
1            72921  100647  31.05.01 Лечебное дело  2015-2016   3.0      5.0   
2            96438  199311   31.05.03 Стоматология  2019-2020   1.0      1.0   
3           273044  162002     38.03.02 Менеджмент  2017-2018   2.0      3.0   
4           371309  108384  31.05.01 Лечебное дело  2018-2019   5.0      9.0   
...            ...     ...                     ...        ...   ...      ...   
105053      224856  176141  40.03.01 Юриспруденция  2017-2018   1.0      2.0   
105054      260998  182794  31.05.01 Лечебное дело  2018-2019   2.0      4.0   
105055       57245  162391  31.05.01 Лечебное дело  2016-2017   1.0      2.0   
105056      218529  202479      38.03.01 Экономика  2017-2018   1.0      1.0   
105057      233318  182325  08.03.01 Строительство  2018-2019   2.0      3.0   

                      АТТЕСТАЦИЯ  \
0                        Экзамен   
1                        Экзамен   
2                        Экзамен   
3                        Экзамен   
4                        Экзамен   
...                          ...   
105053           Курсовая работа   
105054                   Экзамен   
105055  Дифференцированный зачет   
105056  Дифференцированный зачет   
105057           Курсовая работа   

                                               ДИСЦИПЛИНА   index  mark  
0                                               Маркетинг   51081     5  
1       Пропедевтика внутренних болезней, лучевая диаг...   72921     5  
2                                                   Химия   96438     5  
3                                              Статистика  273044     4  
4                                               Педиатрия  371309     3  
...                                                   ...     ...   ...  
105053                              Конституционное право  224856     3  
105054                              Нормальная физиология  260998     3  
105055                                   Иностранный язык   57245     4  
105056                                        Информатика  218529     4  
105057           Технологические процессы в строительстве  233318     4  

[105058 rows x 10 columns]

In [102]:
studs_info

STD_ID     number Пол Статус Дата выпуска  Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                   1   
1      99112    13998.0   Ж    ВЫП   2019-08-31                   1   
2      99109    13999.0   Ж    ВЫП   2019-08-31                   1   
3      99125    13953.0   Ж    ВЫП   2019-08-31                   1   
4     100676    13950.0   М    ВЫП   2019-08-31                   1   
...      ...        ...  ..    ...          ...                 ...   
5461  100678      978.0   Ж     СТ          NaN                   1   
5462  100663     1021.0   Ж     СТ          NaN                   2   
5463  191630     6447.0   Ж     СТ          NaN                   2   
5464  191625     3765.0   Ж     СТ          NaN                   2   
5465  194978     5245.0   Ж     СТ          NaN                   2   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [103]:
# 3 

train.isna().sum()

Unnamed: 0     0
STD_ID         0
НАПРАВЛЕНИЕ    0
ГОД            0
КУРС           0
СЕМЕСТР        0
АТТЕСТАЦИЯ     0
ДИСЦИПЛИНА     0
index          0
mark           0
dtype: int64

In [104]:
# 4

studs_info['направление (специальность)'].value_counts()

Лечебное дело                                                                870
Юриспруденция                                                                462
Стоматология                                                                 452
Педагогическое образование (с двумя профилями подготовки)                    369
Педагогическое образование                                                   279
Экономика                                                                    226
Строительство                                                                214
Менеджмент                                                                   184
Лингвистика                                                                  136
Специальное (дефектологическое) образование                                  131
Теплоэнергетика и теплотехника                                               126
Журналистика                                                                 118
Конструкторско-технологическ

In [105]:
# 5

woman = studs_info[studs_info['Пол']=='Ж']
woman = woman[woman['направление (специальность)']=='Лечебное дело']
woman

STD_ID     number Пол Статус Дата выпуска  Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                   1   
1      99112    13998.0   Ж    ВЫП   2019-08-31                   1   
2      99109    13999.0   Ж    ВЫП   2019-08-31                   1   
3      99125    13953.0   Ж    ВЫП   2019-08-31                   1   
5      99099    13959.0   Ж    ВЫП   2019-08-31                   1   
...      ...        ...  ..    ...          ...                 ...   
5461  100678      978.0   Ж     СТ          NaN                   1   
5462  100663     1021.0   Ж     СТ          NaN                   2   
5463  191630     6447.0   Ж     СТ          NaN                   2   
5464  191625     3765.0   Ж     СТ          NaN                   2   
5465  194978     5245.0   Ж     СТ          NaN                   2   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
5                 Д  31.05.01               Лечебное дело  8800.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
5     2013-06-21 00:00:00     МОУ СОШ (полная), ХОЛМ        2013   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
5               06           21  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[564 rows x 16 columns]

In [106]:
# 6

def my_split_year(line):
  if type(line) == str:
    return line.split('-')[0]
  return line

def my_split_month(line):
  if type(line) == str:
    try:
      return line.split('-')[1]
    except:
      return line
  return line

def my_split_day(line):
  if type(line) == str:
    try:
      mass = line.split(' ')[0]
      return mass.split('-')[2]
    except:
      return line
  return line

# 2013-06-26 00:00:00
studs_info['Год выпуска'] = studs_info['Дата выдачи'].apply(my_split_year)
studs_info['Месяц выпуска'] = studs_info['Дата выдачи'].apply(my_split_month)
studs_info['День выпуска'] = studs_info['Дата выдачи'].apply(my_split_day)
studs_info

STD_ID     number Пол Статус Дата выпуска  Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                   1   
1      99112    13998.0   Ж    ВЫП   2019-08-31                   1   
2      99109    13999.0   Ж    ВЫП   2019-08-31                   1   
3      99125    13953.0   Ж    ВЫП   2019-08-31                   1   
4     100676    13950.0   М    ВЫП   2019-08-31                   1   
...      ...        ...  ..    ...          ...                 ...   
5461  100678      978.0   Ж     СТ          NaN                   1   
5462  100663     1021.0   Ж     СТ          NaN                   2   
5463  191630     6447.0   Ж     СТ          NaN                   2   
5464  191625     3765.0   Ж     СТ          NaN                   2   
5465  194978     5245.0   Ж     СТ          NaN                   2   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [107]:
# 7

len(studs_info['Категория обучения'].unique())

3

In [108]:
# 8

keys = {}
for i, item in enumerate(studs_info['Категория обучения'].unique()):
  keys[item] = i + 1

studs_info['Категория обучения'] = studs_info['Категория обучения'].apply(lambda x: keys[x])

studs_info

STD_ID     number Пол Статус Дата выпуска  Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                   1   
1      99112    13998.0   Ж    ВЫП   2019-08-31                   1   
2      99109    13999.0   Ж    ВЫП   2019-08-31                   1   
3      99125    13953.0   Ж    ВЫП   2019-08-31                   1   
4     100676    13950.0   М    ВЫП   2019-08-31                   1   
...      ...        ...  ..    ...          ...                 ...   
5461  100678      978.0   Ж     СТ          NaN                   1   
5462  100663     1021.0   Ж     СТ          NaN                   2   
5463  191630     6447.0   Ж     СТ          NaN                   2   
5464  191625     3765.0   Ж     СТ          NaN                   2   
5465  194978     5245.0   Ж     СТ          NaN                   2   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [109]:
# 9

studs_info.drop(['Форма обучения'], axis=1)
studs_info

STD_ID     number Пол Статус Дата выпуска  Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                   1   
1      99112    13998.0   Ж    ВЫП   2019-08-31                   1   
2      99109    13999.0   Ж    ВЫП   2019-08-31                   1   
3      99125    13953.0   Ж    ВЫП   2019-08-31                   1   
4     100676    13950.0   М    ВЫП   2019-08-31                   1   
...      ...        ...  ..    ...          ...                 ...   
5461  100678      978.0   Ж     СТ          NaN                   1   
5462  100663     1021.0   Ж     СТ          NaN                   2   
5463  191630     6447.0   Ж     СТ          NaN                   2   
5464  191625     3765.0   Ж     СТ          NaN                   2   
5465  194978     5245.0   Ж     СТ          NaN                   2   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [133]:
# 10

def my_split(line):
  if type(line) == str:
    return line.split()[0]
  return line

'''
man = studs_info[studs_info['Пол']=='М']
man = man[man['Год выпуска']=='2019']
man = man[man['Месяц выпуска']=='08']
man = man[man['День выпуска']=='31']
'''
man = studs_info[studs_info['Пол']=='М'] 
man2 = man['Дата выдачи'].apply(lambda x: my_split(x))
len(man2[man2=='2019-08-31'])

2016-06-24    113
2018-06-30     78
2018-06-29     64
2017-06-30     62
2016-06-23     62
             ... 
2017-09-05      1
2006-06-23      1
2001-06-25      1
1998-06-18      1
2000-07-01      1
Name: Дата выдачи, Length: 440, dtype: int64

## Вариант 2.

Кто делает:
1. Ивлиев Степан Дмитриевич
2. Игнатов Игорь Дмитриевич
3. Казанов Амир Ринатович
4. Клячин Иван Николаевич
5. Котов Александр Андреевич
6. Мелешкин Кирилл Андреевич
7. Файзуллин Илья Денисович
8. Чупрова Ася Александровна

1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в датафрейме studs_info.
4. Посчитать сколько мужчин, сколько женщин в датафрейме studs_info.
5. Создать новый датафрейм из studs_info, в котором будут только мужчины 2019-06-30 года выпуска.
6. Разложить колонку "Дата выдачи" на четыре: Год выдачи, Месяц выдачи, День выдачи, Время выдачи.
7. Посчитать сколько уникальных значений в колонке studs_info['Статус'] и сколько раз каждое значение встречается в датафрейм.
8. Посчитать сколько раз какая оценка встречатся в датафрейм y_train.
9. Посчитать средний балл для студента с ID=175711 по всем предметам.
10. Удалить любую ненужны колонку на ваш вгляд из датафрейм train.

In [ ]:
# Ваше решение

df[df['name'] == 'Ivan']

## Вариант 3.

Кто делает:
1. Морозов Максим Олегович
2. Переттруов Никита Артемович
3. Пославский Егор Константинович
4. Садохин Константин Павлович
5. Скиданов Дмитрий Артёмович
6. Судариков Андрей Геннадьевич
7. Таланов Максим Андреевич

Задания:
1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в датафрейме studs_info.
4. Посчитать сколько уникальных значений в колонке АТТЕСТАЦИЯ и сколько раз каждое из них встречается.
5. Переименовать все колонки датафрейм train, чтобы названия были маленькими буквами.
6. Разложить колонку "направление" на две: Код направления, Название направления.
7. Создать новый датафрейм, в котором будут только мужчины 3 курса обучения.
8. Заменить все строковые значения в колонке Пол на числовые.
9. Сколько всего студентов сдавали предмет "Информатика" в виде экзамена? И какой средний балл по всем сдававшим?

In [ ]:
# Ваше решение